libsel4test/CMakeLists.txt

```diff
@@ -38,6 +38,12 @@ file(GLOB deps src/*.c)
 list(SORT deps)
 
 add_library(sel4test STATIC EXCLUDE_FROM_ALL ${deps})
+if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
```

Comment:
What happens if you compile with GCC but link with ld.lld or compile with Clang but link with ld.bfd?